src/core/server/http/router/request.ts

```diff
@@ -211,15 +211,21 @@ export class KibanaRequest<
 
   private getRouteInfo(request: Request): KibanaRequestRoute<Method> {
     const method = request.method as Method;
-    const { parse, maxBytes, allow, output } = request.route.settings.payload || {};
-    const timeout = request.route.settings.timeout?.socket;
+    const { parse, maxBytes, allow, output, timeout: payloadTimeout } =
+      request.route.settings.payload || {};
 
+    // net.Socket#timeout isn't documented, yet, and isn't part of the types... https://github.com/nodejs/node/pull/34543
+    // the socket is also undefined when using @hapi/shot, or when a "fake request" is used
```

Comment:
It bothers me that the socket is undefined, but the types don't reflect that... We end up replication that with `KibanaRequest#socket`. @elastic/kibana-platform, do you all think we should consider making `KibanaRequest#socket` undefined and using that here?

Reply:
Or we just get rid of the route info's `timeout` property entirely. It's only being used for unit tests, which give us very little certainty that this is actually working...

Reply:
I agree, this is not ideal. I had to do something similar in [`9028c0a` (#71019)](https://github.com/elastic/kibana/pull/71019/commits/9028c0a3609fa2bbffd365c2d8ece7f7a09e284f).

> do you all think we should consider making `KibanaRequest#socket` undefined and using that here?

I'm not sure I love this idea. It will force consumers to handle a situation that is somewhat artificial and should never actually be seen in production. I'd prefer we either remove the `timeout` property or have the default values returned when no real socket is present. The latter option allows us to still have some level of unit testing (even if somewhat superficial) while being able to abstract the socket undefined testing quirk away from consumers.

Reply:
> have the default values returned when no real socket is present

I can do so, but it doesn't feel like it's improving the current situation... The idle socket timeout is currently being returned as part of the "route info". Wouldn't it be rather confusing if someone wrote a test which defined a route with an explicit idle socket timeout, and then used `server.inject` as part of their test, to see the route info return the default idle socket timeout for all routes?